packages/cli/src/ceramic-daemon.ts

```diff
@@ -215,89 +143,43 @@ class CeramicDaemon {
+      app.postAsync(toApiPath('/documents'), this.createReadOnlyDocFromGenesis.bind(this))
+      app.postAsync(toApiPath('/commits'),  this._notSupported.bind(this))
+      app.postAsync(toApiPath('/records'),  this._notSupported.bind(this))
+      app.postAsync(toApiPath('/pins/:docid'),  this._notSupported.bind(this))
+      app.deleteAsync(toApiPath('/pins/:docid'),  this._notSupported.bind(this))
     }
   }
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-  _buildHttpLog (requestStart: number, req: Request, res: Response, extra?: any): HttpLog {
-    const { headers, httpVersion, method, socket, url } = req;
-    const { remoteAddress, remoteFamily } = socket;
-    const httpLog: HttpLog = {
-      request:{
-        timestamp: Date.now(),
-        headers,
-        httpVersion,
-        method,
-        remoteAddress,
-        remoteFamily,
-        url
-      }
-    }
-    const now = Date.now()
-    httpLog.response = {
-      timestamp: now,
-      processingTime: now - requestStart,
-      body: extra && extra.body || null,
-      statusCode: res.statusCode,
-      statusMessage: res.statusMessage,
-      requestError: extra && extra.requestError || null
-    }
-    return httpLog
-  }
-
-  async healthcheck (req: Request, res: Response, next: NextFunction): Promise<void> {
-    const freeCpu: any = await new Promise((resolve) => cpuFree(resolve))
-    const cpuUsage: number = 1 - freeCpu
-
-    const freeMemory = freememPercentage()
-    const memUsage: number = 1 - freeMemory
-
-    const stats = `maxHealthyCpu=${this.maxHealthyCpu} cpuUsage=${cpuUsage} freeCpu=${freeCpu} maxHealthyMemory=${this.maxHealthyMemory} memoryUsage=${memUsage} freeMemory=${freeMemory}`
-
-    if (cpuUsage > this.maxHealthyCpu || memUsage > this.maxHealthyMemory) {
-      this.logger.debug(stats)
-      res.status(503)
-      return next(new Error('Ceramic failed a healthcheck. Insufficient resources.'))
-    } else {
-      res.status(200).send('Alive!')
-    }
+  healthcheck (req: Request, res: Response): void {
+    res.status(200).send('Alive!')
   }
 
   /**
    * Create document from genesis commit
    * @dev Useful when the docId is unknown, but you have the genesis contents
    */
-  async createDocFromGenesis (req: Request, res: Response, next: NextFunction): Promise<void> {
+  async createDocFromGenesis (req: Request, res: Response): Promise<void> {
     const { doctype, genesis, docOpts } = req.body
-    try {
-      const doc = await this.ceramic.createDocumentFromGenesis(doctype, DoctypeUtils.deserializeCommit(genesis), docOpts)
-      res.json({ docId: doc.id.toString(), state: DoctypeUtils.serializeState(doc.state) })
-    } catch (e) {
-      return next(e)
-    }
-    next()
+    const doc = await this.ceramic.createDocumentFromGenesis(doctype, DoctypeUtils.deserializeCommit(genesis), docOpts)
+    res.json({ docId: doc.id.toString(), state: DoctypeUtils.serializeState(doc.state) })
```

Comment:
Here as an example, can we do a check to ensure that `doctype`, `gensis`, and `docOpts` are not empty? Or is that already handled by the methods below?

Also is there any potential attack vector here we can mitigate with sanitization?

Reply:
This kind of checks would eliminate some execution steps required to achieve same checks in the underlying code, which is good. How good? Have no data. Probably, minuscule, but this is just a guess. I am pretty sure with error-to-http mapping, UX/DX is the same, with a benefit of having a single place where the checks are executed.

Reply:
Ok let's punt for now